core/lib.rs

```diff
@@ -428,7 +424,56 @@ impl Database {
         self.open_flags.contains(OpenFlags::ReadOnly)
     }
 
-    fn init_pager(&self, page_size: Option<usize>) -> Result<Pager> {
+    /// If we do not have a physical WAL file, but we know the database file is initialized on disk,
+    /// we need to read the page_size from the database header.
+    fn read_page_size_from_db_header(&self) -> Result<PageSize> {
+        turso_assert!(
+            self.db_state.is_initialized(),
+            "read_page_size_from_db_header called on uninitialized database"
+        );
+        let buf = Arc::new(Buffer::new_temporary(PageSize::MIN as usize)); // this needs to be a block multiple for O_DIRECT
```

Comment:
NIT: assert that `PageSize::MIN % 512 == 0`? Or what is the block size for `O_DIRECT`?